housingbayarea/bloom
# feat: resource translation scripts

# Pull Request Template

## Issue Overview

This PR addresses #2782


## Description

This adds export and import scripts for resources. This will go in core, but I opened a PR created from the Alameda branch, since it'll be easier to test with the resources there.

## How Can This Be Tested/Reviewed?

Check out this locally and from the root directory of bloom, you can run `ts-node sites/public/scripts/resource-translations-export.ts`. This will create a CSV file. Then, if you run `ts-node sites/public/scripts/resource-translations-import.ts sites/public/scripts/resource-translations-export.csv`, it will import the CSV. If this is successful, you should see no changed files.

If you delete one of the resource md files or clear its contents, running the import should restore it.

If you modify the CSV and run the import, you should see those changes reflected in the appropriate files.

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
